hasktorch/src/Torch/Typed/Tensor.hs

```diff
@@ -599,3 +665,43 @@ toDouble t = D.asValue . toDynamic . toCPU $ t
 
 toBool :: forall device. Tensor device 'D.Bool '[] -> Bool
 toBool t = D.asValue . toDynamic . toCPU $ t
+
+--------------------------------------------------------------------------------
+-- NamedTensor
+--------------------------------------------------------------------------------
+
+data NamedTensor (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: Shape) where
+  FromTensor :: forall device dtype shape' shape. shape ~ ToNats shape' => Tensor device dtype shape ->  NamedTensor device dtype shape'
+
+instance Unnamed (NamedTensor device dtype shape) where
```

Comment:
do you have an instance for `Tensor device dtype shape`, too?

Reply:
Yes. [This code](https://github.com/hasktorch/hasktorch/pull/536/files/2ab534287bd778c9325ead060f1a878fde0453da#diff-5757ffa202fb7ecd16a6aa1c8217948a432d109d10ae364fdb8963f6e2303d80R162) is the instance.